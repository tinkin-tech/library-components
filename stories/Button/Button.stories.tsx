import * as React from 'react'
import { action } from '@storybook/addon-actions'

import Button from './Button'

export default {
  title: 'Button',
  component: Button,
}

export const Active = (): React.ReactNode => {
  return <Button text="Click" action={action('clicked')} type="primary" />
}

export const Disable = (): React.ReactNode => (
  <Button
    text="Click"
    action={action('clicked')}
    type="primary"
    disable={true}
  />
)

export const stylesButton = (): React.ReactNode => (
  <div id="group-buttons">
    <div className="p-b-lg">
      <div className="size-5">Small</div>
      <Button
        text="Click"
        action={action('clicked')}
        type="primary"
        typeButton="small"
      />
    </div>
    <div className="p-b-lg">
      <div className="size-5">Text</div>
      <Button
        text="Click"
        action={action('clicked')}
        type="primary"
        typeButton="text"
      />
    </div>
    <div className="p-b-lg">
      <div className="size-5">Border</div>
      <Button
        text="Click"
        action={action('clicked')}
        type="primary"
        typeButton="border"
      />
    </div>
  </div>
)

export const typesButton = (): React.ReactNode => (
  <div id="group-buttons">
    <div className="p-b-lg">
      <div className="size-5">Primary</div>
      <Button text="Click" action={action('clicked')} type="primary" />
    </div>
    <div className="p-b-lg">
      <div className="size-5">Danger</div>
      <Button text="Click" action={action('clicked')} type="danger" />
    </div>
    <div className="p-b-lg">
      <div className="size-5">Info</div>
      <Button text="Click" action={action('clicked')} type="info" />
    </div>
    <div className="p-b-lg">
      <div className="size-5">White</div>
      <Button text="Click" action={action('clicked')} type="white" />
    </div>
  </div>
)

export const withIcon = (): React.ReactNode => {
  return <Button text="Click" action={action('clicked')} type="primary" />
}

export const withIconDisable = (): React.ReactNode => (
  <Button
    text="Click"
    action={action('clicked')}
    type="primary"
    disable={true}
  />
)

export const withIconStylesButton = (): React.ReactNode => (
  <div id="group-buttons">
    <div className="p-b-lg">
      <div className="size-5">Small with icon</div>
      <Button
        text="Click"
        action={action('clicked')}
        type="primary"
        typeButton="small"
      />
    </div>
    <div className="p-b-lg">
      <div className="size-5">Text with icon</div>
      <Button
        text="Click"
        action={action('clicked')}
        type="primary"
        typeButton="text"
      />
    </div>
    <div className="p-b-lg">
      <div className="size-5">Border with icon</div>
      <Button
        text="Click"
        action={action('clicked')}
        type="primary"
        typeButton="border"
      />
    </div>
  </div>
)

export const withIconTypesButton = (): React.ReactNode => (
  <div id="group-buttons">
    <div className="p-b-lg">
      <div className="size-5">Primary</div>
      <Button text="Click" action={action('clicked')} type="primary" />
    </div>
    <div className="p-b-lg">
      <div className="size-5">Danger</div>
      <Button text="Click" action={action('clicked')} type="danger" />
    </div>
    <div className="p-b-lg">
      <div className="size-5">Info</div>
      <Button text="Click" action={action('clicked')} type="info" />
    </div>
    <div className="p-b-lg">
      <div className="size-5">White</div>
      <Button text="Click" action={action('clicked')} type="white" />
    </div>
  </div>
)
