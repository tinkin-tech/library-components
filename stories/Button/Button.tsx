import * as React from 'react'

interface PropsInterface {
  text: string
  action: () => void
  type: 'primary' | 'danger' | 'white' | 'info'
  typeButton?: 'border' | 'text' | 'small'
  className?: string
  disable?: boolean
}

const Button = (props: PropsInterface): React.ReactElement<PropsInterface> => {
  const { type, text, action, className, disable, typeButton } = props
  return (
    <a
      className={`button-component flex-row flex-no-wrap flex-middle ${
        disable ? 'disable' : ''
      } type-${type} ${typeButton}-button ${className || ''}`}
      data-testid="button-component"
      onClick={disable ? (): null => null : action}
    >
      <span className="flex-column">{text}</span>
    </a>
  )
}

export default React.memo(Button)
