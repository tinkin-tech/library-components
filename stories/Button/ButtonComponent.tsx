import * as React from 'react'

type buttonType =
  | 'primary'
  | 'secondary'
  | 'warning'
  | 'danger'
  | 'info'
  | 'success'
  | 'border-primary'
  | 'border-secondary'
  | 'border-warning'
  | 'border-danger'
  | 'border-info'
  | 'border-success'

interface IButtonComponent {
  buttonText: string
  onClick?: () => void
  disable?: boolean
  visualDisable?: boolean
  buttonType?: buttonType
}

const ButtonComponent = (props: IButtonComponent): React.ReactElement => {
  const {
    buttonText,
    onClick,
    disable,
    visualDisable,
    buttonType = 'primary',
  } = props

  return (
    <button
      value={buttonText}
      onClick={disable ? null : onClick}
      className={`${visualDisable ? 'disable' : ''} ${buttonType}`}
    />
  )
}

export default ButtonComponent
