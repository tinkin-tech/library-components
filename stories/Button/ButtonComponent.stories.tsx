import * as React from 'react'

import { IButtonComponent, ButtonComponent } from './ButtonComponent'

export default {
  title: 'Button',
  component: ButtonComponent,
}

const Template = (arg: IButtonComponent): JSX.Element => (
  <ButtonComponent {...arg} />
)

export const Default: { args: IButtonComponent } = Template.bind({})

Default.args = {
  buttonText: 'Buton Content',
}
