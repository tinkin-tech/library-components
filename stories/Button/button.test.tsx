import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import Button from './Button';

describe('Button test', () => {
  it('Should take snapshot of button component with all require properties',
     async () => {
    const component = renderer.create(
      <Button
        type="danger"
        disable={false}
        text="Boton de prueba"
        action={null}
        typeButton="border"
        />
    )
    let componentJSON = component.toJSON();
    expect(componentJSON).toMatchSnapshot();
  });
  it('Should call action function one time when click button', async () => {
    const action = jest.fn(() => null);
    render(<Button type="danger" disable={false} text="Boton de prueba" action={action} typeButton="border" />);
    fireEvent.click(screen.getByTestId('button-component'));
    expect(action).toBeCalledTimes(1);
  });
});
