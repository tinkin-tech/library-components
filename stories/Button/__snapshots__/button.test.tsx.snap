// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Should take snapshot of button component with all require properties 1`] = `
<a
  className="button-component flex-row flex-no-wrap flex-middle 
             type-danger border-button "
  onClick={null}
>
  <span
    className="flex-column"
  >
    Boton de prueba
  </span>
</a>
`;
